volume-providers/vpc/provider/util.go

```diff
@@ -86,6 +86,40 @@ func skipRetry(err *models.Error) bool {
 	return false
 }
 
+// FlexyRetry ...
+func FlexyRetry(logger *zap.Logger, retryfunc func() (interface{}, error), skipRetryFunc func(interface{}, *models.Error) bool) error {
+	var err error
+
+	for i := 0; i < maxRetryAttempt; i++ {
+		if i > 0 {
+			time.Sleep(time.Duration(retryGap) * time.Second)
+		}
+		obj, err := retryfunc()
+		if err != nil {
+			//Skip retry for the below type of Errors
+			modelError, ok := err.(*models.Error)
+			if !ok {
+				continue
+			}
+			if skipRetryFunc(obj, modelError) {
```

Comment:
Caller may or may not pass this skipRetryFunc . You can check for `nil`. If its `nil` lets call default skipErrorCode logic which will be applicable for most of the callers .

Reply:
we discussed on this